homeassistant/components/history/__init__.py

```diff
@@ -210,7 +210,7 @@ def states_to_json(
     each list of states, otherwise our graphs won't start on the Y
     axis correctly.
     """
-    result = defaultdict(list)
+    result = {ent_id: [] for ent_id in entity_ids}
```

Comment:
Insertion order is only maintained in 3.6+, otherwise an `OrderedDict` needs to be used.  While we're at it, why not also sort the `entity_ids`?

Reply:
Home Assistant version 0.97 requires python 3.6 or later.
`entity_ids` can be specified by the user, including ordering (see the linked issue and the frontend PR for examples).